### PR TITLE
[PYTHON] Properties API, improvements of OVAny

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - Submodules reflect structure of C++ namespaces as close as possible.
 - RO properties return string with the name of property.
 - RW properties without argument return the name of property, when matching argument is passed - return a tuple with the name and value.
 
Importing:
 ```python
>>> from openvino.runtime import properties
```

RO properties works only with call:
```python
>>> properties.available_devices()
'AVAILABLE_DEVICES'

>>> properties.available_devices(777)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: available_devices(): incompatible function arguments. The following argument types are supported:
    1. () -> str

Invoked with: 777
```

RW properties works with matching argument, here Affinity has an addition of enum values:
```python
>>> properties.affinity()
'AFFINITY'
>>> properties.affinity(properties.Affinity.NONE)
('AFFINITY', NONE)
```

When creating properties dictionary, values can be passed in three ways:
```python
# Dict from tuples
dict([
    properties.enable_profiling(True),
    properties.affinity(properties.Affinity.NONE),
    properties.hint.inference_precision(Type.f32),
    properties.hint.performance_mode(properties.hint.PerformanceMode.LATENCY),
    properties.hint.num_requests(12),
])

# Pure dict of names to values
{
    properties.enable_profiling(): True,
    properties.affinity(): properties.Affinity.NONE,
    properties.hint.inference_precision(): Type.f32,
    properties.hint.performance_mode(): properties.hint.PerformanceMode.LATENCY,
    properties.hint.num_requests(): 12,
}

# Mixed dict
{
    properties.enable_profiling(): True,
    "AFFINITY": "NONE",
    "INFERENCE_PRECISION_HINT": Type.f32,
    properties.hint.performance_mode(): properties.hint.PerformanceMode.LATENCY,
    properties.hint.num_requests(): 12,
}
```

### Tickets:
 - *81827*

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp [hunk 1]
```diff
@@ -164,10 +161,10 @@ void regclass_CompiledModel(py::module m) {
 
     cls.def(
         "get_property",
-        [](ov::CompiledModel& self, const std::string& name) -> py::object {
-            return Common::utils::from_ov_any(self.get_property(name));
+        [](ov::CompiledModel& self, const py::object& property) -> py::object {
+            return Common::utils::property_as_py_object<ov::CompiledModel>(self, property);
         },
-        py::arg("name"),
+        py::arg("property"),
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/core/compiled_model.cpp:167]: won't it break compatibility?

File: src/bindings/python/src/pyopenvino/core/core.cpp [hunk 2]
```diff
@@ -69,6 +63,24 @@ void regclass_Core(py::module m) {
             :type properties: dict
         )");
 
+    cls.def(
+        "get_property",
+        [](ov::Core& self, const std::string& device_name, const py::object& property) -> py::object {
+            return Common::utils::property_as_py_object<ov::Core>(self, device_name, property);
+        },
+        py::arg("device_name"),
+        py::arg("property"),
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/core/core.cpp:72]: won't it break compatibility?

File: src/bindings/python/src/pyopenvino/utils/utils.cpp [hunk 3]
```diff
@@ -146,10 +95,39 @@ py::object from_ov_any(const ov::Any& any) {
...
+            properties_to_cpp[property.first.cast<std::string>()] = ov::Any(py_object_to_any(property.second));
+        } else {
+            py::print("Property is broken and unusable!");
+            // TODO: raise error!
+        }
+    }
+    return properties_to_cpp;
+}
 };  // namespace utils
 };  // namespace Common
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/utils/utils.cpp:133]: please, add blank line

Comment (jiwaszki) [reply src/bindings/python/src/pyopenvino/utils/utils.cpp:133]: Will do!

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp (hunk 1 above)
Comment (jiwaszki) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp:167]: That is a very good question -- should we keep "name" just for sake of compatibility in kwagrs? New argument name is matching current `Union[str, Property]` possibility. Awaiting your feedback!

CC @AlexeyLebedev1

Comment (jiwaszki) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp:166]: @akuporos we decided to change it if possible and see if it breaks any of CIs. `property/properties` are more meaningful than simple `name`.

Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp:166]: I agree, but from my perspective it's a little bit late change it as our users may use named argument `name` in his/her code.

File: src/bindings/python/src/pyopenvino/utils/utils.cpp [hunk 4]
```diff
@@ -24,117 +23,67 @@ py::object from_ov_any(const ov::Any& any) {
...
     }
     // Check for std::map<std::string, int>
     else if (any.is<std::map<std::string, int>>()) {
-        auto val = any.as<std::map<std::string, int>>();
-        PyObject* dict = PyDict_New();
-        for (const auto& it : val) {
-            PyDict_SetItemString(dict, it.first.c_str(), PyLong_FromLong((long)it.second));
-        }
-        return py::cast<py::object>(dict);
+        return py::cast(any.as<std::map<std::string, int>>());
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/utils/utils.cpp:86]: really appreciate these simplifications!

Comment (jiwaszki) [reply src/bindings/python/src/pyopenvino/utils/utils.cpp:86]: No problem!

File: src/bindings/python/src/pyopenvino/core/core.cpp [hunk 5]
```diff
@@ -65,22 +70,59 @@ void regclass_Core(py::module m) {
...
     cls.def(
         "compile_model",
         [](ov::Core& self,
            const std::shared_ptr<const ov::Model>& model,
            const std::string& device_name,
-           const std::map<std::string, std::string>& properties) {
+           const std::map<std::string, py::object>& properties) {
+            auto _properties = Common::utils::properties_to_any_map(properties);
+            py::gil_scoped_release release;
             return self.compile_model(model, device_name, {properties.begin(), properties.end()});
```
Comment (AlexeyLebedev1) [inline src/bindings/python/src/pyopenvino/core/core.cpp:121]:
```suggestion
            return self.compile_model(model, device_name, _properties);
```

File: src/bindings/python/tests/test_inference_engine/test_properties.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
+# Copyright (C) 2018-2022 Intel Corporation
```
Comment (AlexeyLebedev1) [inline src/bindings/python/tests/test_inference_engine/test_properties.py:1]:
```suggestion
# Copyright (C) 2022 Intel Corporation
```

**Review by AlexeyLebedev1 — CHANGES_REQUESTED** [review]

**Review by AlexeyLebedev1 — APPROVED** [review]

**Pull request merged**